src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -291,6 +299,17 @@ def _add_defaults_to_definition(original_definition: Definition) -> Definition:
     return definition_with_defaults
 
 
+def _update_metrics(template_env: TemplatedEnvironment, definition: Definition):
+    metrics = get_cli_context().metrics
+
+    # render_definition_template is invoked multiple times both by the user
+    # and by us so we should make sure we don't overwrite a 1 with a 0 here
+    metrics.increment_counter(CLICounterField.PDF_TEMPLATES, 0)
```

Comment:
I don't understand this line, for me do nothing. Am I wrong?

Reply:
See https://github.com/snowflakedb/snowflake-cli/pull/1597#discussion_r1771667230 for more detailed explanation, but we are essentially marking the feature as available, but not used

Reply:
Marcus, I'm not sure about that. This is an _increment_ call with value 0, so it really looks odd. If you are trying to use this as a "create if not exists" call, maybe introduce something other than increment.